susemanager-utils/susemanager-sls/src/modules/mgrnet.py

```diff
@@ -0,0 +1,90 @@
+"""
+Module for gathering DNS FQDNs
+"""
+
+import logging
+import re
+import time
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import salt.utils.network
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    """
+    Only works on POSIX-like systems having 'host' or 'nslookup' available
+    """
+    if not (__utils__["path.which"]("host") or __utils__["path.which"]("nslookup")):
+        return (False, "Neither 'host' nor 'nslookup' is available on the system")
+    return True
+
+
+def dns_fqdns():
+    """
+    Return all known DNS FQDNs for the system by enumerating all interfaces and
+    then trying to reverse resolve them with native DNS tools
+    """
+    # Provides:
+    # dns_fqdns
+
+    grains = {}
+    fqdns = set()
+    cmd_run_all_func = __salt__["cmd.run_all"]
+    if __utils__["path.which"]("host"):
+        cmd = "host"
+        cmd_ret_regex = re.compile(r".* domain name pointer (.*)\.$")
+    elif __utils__["path.which"]("nslookup"):
+        cmd = "nslookup"
+        cmd_ret_regex = re.compile(r".*\tname = (.*)\.$")
+    else:
+        log.error("Neither 'host' nor 'nslookup' is available on the system")
+        return {"dns_fqdns": []}
```

Comment:
nitpick: Given the check in `__virtual__`, I guess this is dead code now?

Reply:
Yes, looks redundant, but there are still some chances that `host` and `nslookup` could be removed between the moments of invoking `__virtual__` and `dns_fqdns`, so I feel safer if leave it as it is if you don't mind.